### PR TITLE
[FEATURE] Define possible field names as chain of fields

### PR DESCRIPTION
**Pull request by websi**
Use TypoScript "getData" feature.

### CODE REVIEW

File: Classes/ViewHelpers/KeyValueViewHelper.php [hunk 1]
```diff
@@ -2,37 +2,44 @@
...
 /**
  * Class KeyValueViewHelper
  * @package YoastSeoForTypo3\YoastSeo\ViewHelpers
  */
 class KeyValueViewHelper extends AbstractViewHelper
 {
+
+    /**
+     * @var \TYPO3\CMS\Extbase\Configuration\ConfigurationManagerInterface
+     * @inject
```
Comment (haassie) [inline Classes/ViewHelpers/KeyValueViewHelper.php]: Can you use a setter injection for this in stead of a property injection?

File: Configuration/TypoScript/constants.txt [hunk 2]
```diff
@@ -1,7 +1,7 @@
 plugin.tx_yoastseo {
-	titleField = tx_yoastseo_title
-	descriptionField = description
-	canonicalTagField = tx_yoastseo_canonical_url
+	titleField = TSFE:altPageTitle // field:tx_yoastseo_title
```
Comment (haassie) [inline Configuration/TypoScript/constants.txt]: I didn't got this working?

**Comment by websi** [comment]
> This is a breaking change because configuration is changed. What's in your opinion the reason of changing this?

Yes, this is a breaking change.
This change add the possibility, to change the content of the page title without change the template.
Additional you can use a alternate field or register from your extension to manipulate the content of page title.

For example you can use:

```
titleField = register:myRegisterValue // TSFE:altPageTitle // field:tx_yoastseo_title
```

This add more flexibility to change the title tag for an integrator.

**Comment by haassie** [comment]
In version 2 the setup for defining the title field is a little different. In this version the viewhelper is even gone. Therefor I will close this PR.

**Pull request closed without merging**